iceoryx2/src/node/mod.rs

```diff
@@ -0,0 +1,709 @@
+        let msg = "Unable to read node details";
+
+        if node_storage
+            .read(unsafe { read_content.as_mut_vec() }.as_mut_slice())
+            .is_err()
+        {
+            fail!(from origin, with NodeReadStorageFailure::ReadError,
+                "{} since the content of the node config storage could not be read.", msg);
+        }
+
+        let node_details = fail!(from origin,
+                    when Service::ConfigSerializer::deserialize::<NodeDetails>(unsafe { read_content.as_mut_vec()}),
+                    with NodeReadStorageFailure::Corrupted,
+                "{} since the contents of the node config storage is corrupted.", msg);
+
+        Ok(Some(node_details))
+    }
+}
+
+/// Creates a [`Node`].
+///
+/// ```
+/// use iceoryx2::prelude::*;
+///
+/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let node = NodeBuilder::new()
+///                 .name(&NodeName::new("my_little_node")?)
+///                 .create::<zero_copy::Service>()?;
+///
+/// // do things with your cool new node
+/// # Ok(())
+/// # }
+/// ```
+#[derive(Debug, Default)]
+pub struct NodeBuilder {
+    name: Option<NodeName>,
+    config: Option<Config>,
+}
+
+impl NodeBuilder {
+    /// Creates a new [`NodeBuilder`]
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Sets the [`NodeName`] of the to be created [`Node`].
+    pub fn name(mut self, value: &NodeName) -> Self {
+        self.name = Some(value.clone());
+        self
+    }
+
+    /// Sets the config of the [`Node`] that will be used to create all entities owned by the
+    /// [`Node`].
+    pub fn config(mut self, value: &Config) -> Self {
+        self.config = Some(value.clone());
+        self
+    }
+
+    /// Creates a new [`Node`] for a specific [`service::Service`]. All entities owned by the
+    /// [`Node`] will have the same [`service::Service`].
+    pub fn create<Service: service::Service>(self) -> Result<Node<Service>, NodeCreationFailure> {
+        let msg = "Unable to create node";
+        let node_id = fail!(from self, when UniqueSystemId::new(),
+                                with NodeCreationFailure::InternalError,
+                                "{msg} since the unique node id could not be generated.");
+        let monitor_name = fatal_panic!(from self, when FileName::new(node_id.value().to_string().as_bytes()),
+                                "This should never happen! {msg} since the UniqueSystemId is not a valid file name.");
+        let config = if let Some(ref config) = self.config {
+            config.clone()
+        } else {
+            Config::get_global_config().clone()
+        };
+
+        let (details_storage, details) =
+            self.create_node_details_storage::<Service>(&config, &monitor_name)?;
+        let token = self.create_token::<Service>(&config, &monitor_name)?;
+
+        Ok(Node {
+            id: node_id,
+            monitor: ManuallyDrop::new(token),
```

Comment:
Food for thought: when a `token` is provided, should the member also be called `token`?